barretenberg/cpp/src/barretenberg/ecc/fields/field_declarations.hpp

```diff
@@ -329,8 +329,29 @@ template <class Params_> struct alignas(32) field {
         // if the modulus is a 256-bit integer, we need to use a basis where g1, g2 have been shifted by 2^384
         if constexpr (Params::modulus_3 >= 0x4000000000000000ULL) {
             split_into_endomorphism_scalars_384(k, k1, k2);
-            return;
+        } else {
+            std::pair<std::array<uint64_t, 2>, std::array<uint64_t, 2>> ret =
+                split_into_endomorphism_scalars_no_shift(k);
+            k1.data[0] = ret.first[0];
+            k1.data[1] = ret.first[1];
+
+            // TODO(AD): We should move away from this hack by adapting split_into_endomorphism_scalars_no_shift
+#if !defined(__clang__) && defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Warray-bounds"
+#endif
+            k2.data[0] = ret.second[0]; // NOLINT
+            k2.data[1] = ret.second[1];
+#if !defined(__clang__) && defined(__GNUC__)
+#pragma GCC diagnostic pop
+#endif
         }
+    }
+
+    static std::pair<std::array<uint64_t, 2>, std::array<uint64_t, 2>> split_into_endomorphism_scalars_no_shift(
```

Comment:
Why is it called "no_shift"?

Reply:
I was trying to find a good name as is the counter part to this case technically, but I may be off the mark
<img width="665" alt="Screenshot 2024-02-12 at 1 41 23 PM" src="https://github.com/AztecProtocol/aztec-packages/assets/163993/214eee52-ea23-4153-a2be-6d3b4c3b118f">

Reply:
I'll just call it `split_into_endomorphism_scalars`, though, I think. If it was ever misused the static assert will catch it

Reply:
You can specify that it works for values less than 256 bits if you want, but shift means something else, so it's a bit confusing